planet/src/main/webapp/js/placesItinerary.js

```diff
@@ -52,33 +55,59 @@ function updateStartingAddress() {
 
 export function renderPlaces() {
     displayedPlaces = {};
+    allPlaces = [];
+    placeCount = 0;
     const userId = firebase.auth().currentUser.uid;
     const placesRef = database.ref('users/' + userId + '/places');
-    placesRef.once('value', (placesSnapshot) => {
+    placesRef.orderByValue().once('value', (placesSnapshot) => {
         const placesContainer = document.getElementById('places');
         placesContainer.innerHTML = '';
         const savedPlaces = placesSnapshot.val();
         for (const placeId in savedPlaces) {
-            // Make request with fields
-            let placeRequest = {
-                placeId: placeId,
-                fields: ['place_id','name','rating','formatted_address',
-                    'opening_hours','photos','url']
-            };
-            let service = new google.maps.places.PlacesService(map);
-            service.getDetails(placeRequest, renderPlaceDetailsCallback); 
+            allPlaces.push(placeId);
```

Comment:
Is it possible to just directly assign `allPlaces = savedPlaces` (or copy it with a oneliner, if we need a copy: `allPlaces =  [...savedPlaces]`)

Reply:
`savedPlaces` is a javascript object with placeIds as keys and timestamps as values, however `allPlaces` is an array sorted by timestamp. Therefore i'm not sure if we should changing allPlaces to objects will work with the existing code.

Reply:
Actually, I modified the code a little bit so that the returned data is actually in order (when we call placesSnapshot.val() the order is based on the browser, however if we use a forEach function for the childPlace we can get the correct order)

Reply:
does placesSnapshot.keys() give you the correct ordering?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys

If not, forEach looks good as well.

Reply:
Unfortunately no